### DISCUSSION TITLE
Import not executing batch

### DISCUSSION THREAD

**Discussion by wpas81** [filamentphp/filament#12616] [Help]
### Package

Actions

### Package Version

latest

### How can we help you?

Hi

I've created an import function but it won't execute the job. I've done the following
- create a new laravel project with Jetstream & filament
- create the queue  & notifications tables and the actions-notifications tables as shown in the [docs](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import#overview)
- added ->databasenotifications() to the panel-class in Providers\Filament\AdminPanelProvider.php
- created an importer with the --generate option and added it to the table-header

I can upload a CSV and map the columns. When I hit the import-button it show the "x rows are being processed in the background" notification...But nothing happens. I see the records being added to the jobs & batches tables. But they're not triggered at all. No errors are shown or logged either.

What am I missing here?!

thanks in advance for any advice!

**Comment by micraux** [comment]
Hi,

If you're using Laravel v11, try to [run the queue worker](https://laravel.com/docs/11.x/queues#running-the-queue-worker).

In your terminal: `php artisan queue:work`

**Comment by wpas81** [comment]
yes and that processes the jobs but when there are import errors it keeps retrying them + in the backend no notifications are send (nor the "failed lines csv"-download.
So it seems it's Filament which is not triggering the process  job action

I've added a dd('debug breakpoint') in the importer class > resolveRecord() function but it doesn't get triggered

**Comment by wpas81** [comment]
ok so I've debugged it down to the bus::batch() call in vendor/filament/concerns/CanImportCsv.php  this calls the job. 
In jobs/importCsv.php it runs the __construct-class but the handle()-class is not called....

**Comment by wpas81** [comment]
...found it... I had to start the "php artisan queue:listen" command. I misunderstood the documentation and was under the impression the importCsv-job would trigger the batch execution. My bad :facepalm:

**Reply by ilhamwahyu-ni** [reply]
its still not work
![image](https://github.com/user-attachments/assets/d55e575a-eca5-44b1-be8b-ac9339ce6e3d)

![image](https://github.com/user-attachments/assets/d6453c61-bbdc-4cc2-979b-23599f44b97d)

need help..

**Comment by tfranssen** [comment]
@wpas81 Thanks for your post! I was running into the same issues.

**Reply by wpas81** [reply]
always good to know I'm not the only one :)

**Comment by halilcosdu** [comment]
Hello guys,

I don't want to open a new topic, but I have an issue with exporting. Sometimes, my jobs get stuck in pending. (Memory issue? idk)They don't fail or anything; it just says pending. Do you know how I can fix this?

Using : Horizon


<img width="1187" alt="Screenshot 2024-05-24 at 12 58 47" src="https://github.com/filamentphp/filament/assets/6373017/e67e2488-2daa-4936-ad10-c45887a7e173">

**Reply by abdulmannans-joistic** [reply]
Have you found fix for this?

**Reply by halilcosdu** [reply]
Looks like memory issue and job timeout issue. Try to separate queue to some long-running-queue and increase server memory little bit. One of them should fix the error.

**Reply by abdulmannans-joistic** [reply]
Thank you
